Library/Homebrew/cmd/uninstall.rb

```diff
@@ -9,15 +10,41 @@ def uninstall
       ARGV.kegs.each do |keg|
         keg.lock do
           puts "Uninstalling #{keg}... (#{keg.abv})"
+
+          # Find old name and old name symlinks for formula and remove them
+          begin
+            formula = Formulary.from_rack(keg.rack)
+            if formula.oldname
+              old_opt = HOMEBREW_PREFIX/"opt"/"#{formula.oldname}"
+              old_cellar = HOMEBREW_CELLAR/"#{formula.oldname}"
```

Comment:
There is no entry in `LinkedKegs`, because `LinkedKegs` are used only for installed formulae and linked formula. When we rename software and perform migrations, we unlink the keg and uninstall it.